tpmutil/structures.go

```diff
@@ -95,6 +105,20 @@ func (b *U32Bytes) TPMUnmarshal(in io.Reader) error {
 		return err
 	}
 	size := int(tmpSize)
+
+	// If the reader is a bytes.Buffer, we check there are enough unread
+	// bytes to contain the data section indicated by the size prefix.
+	if bb, ok := in.(*bytes.Buffer); ok {
+		if size > bb.Len() {
+			return io.ErrUnexpectedEOF
+		}
```

Comment:
Similar to above, I also don't think we need this check here either, as bytes.Buffer.Read handles the `io.ErrUnexpectedEOF`. We should just unconditionally refuse to unmarshal if `size > 1MiB`.